src/main/java/edu/greenblitz/tobyDetermined/subsystems/Dashboard.java

```diff
@@ -1,47 +1,173 @@
+				int finalI = i;
+				int finalHeight = height.ordinal();
+				grid.addBoolean(i + 1 + " " + height, () -> (Grid.getInstance().getSelectedHeightID() == finalHeight && Grid.getInstance().getSelectedPositionID() == finalI))
+						.withPosition(finalI, 2 - finalHeight);
+			}
+		}
+
+		//pose
+		ShuffleboardLayout robotPoseWidget = driversTab.getLayout("Robot pose", BuiltInLayouts.kList)
+				.withPosition(0, 2).withSize(1, 2).withProperties(Map.of("Label position", "TOP"));
+		robotPoseWidget.addDouble("X", () -> SwerveChassis.getInstance().getRobotPose().getX());
+		robotPoseWidget.addDouble("Y", () -> SwerveChassis.getInstance().getRobotPose().getY());
+		robotPoseWidget.addDouble("Rotation", () -> SwerveChassis.getInstance().getRobotPose().getRotation().getDegrees());
+
+		//battery
+		driversTab.addDouble("Battery", () -> Battery.getInstance().getCurrentVoltage())
+				.withPosition(9, 3);
+
+		//object inside
+		driversTab.addString("Object inside", /*()->RotatingBelly.getInstance().getGameObject().toString()*/() -> "robot does not exist")
+				.withSize(1, 1).withPosition(2, 2);
+
+		//field
+		driversTab.add("Field", SwerveChassis.getInstance().getField()).withPosition(5, 2).withSize(3, 2);
+
+
+		//console
+		ShuffleboardLayout console =Console.getShuffleboardConsole(driversTab)
+				.withPosition(8,0).withSize(2,3).withProperties(Map.of("Label position", "TOP"));
+
+
+		//ready to place
+		driversTab.addBoolean("Ready to place", () -> false).withPosition(3, 2).withSize(1, 2);
+		//todo check if at place and arm in pos
+
+
+	}
+
+	public void armDashboard() {
+		ShuffleboardTab armTab = Shuffleboard.getTab("Arm debug");
+
+		//arm states
+		ShuffleboardLayout armStateWidget = armTab.getLayout("Arm states", BuiltInLayouts.kGrid)
+				.withPosition(0, 0).withSize(2, 2).withProperties(Map.of("Label position", "TOP", "Number of columns", 2, "Number of rows", 2));
+		armStateWidget.addString("Extender State", () -> Extender.getInstance().getState().toString()).withPosition(0, 0);
+		armStateWidget.addDouble("Length", () -> Extender.getInstance().getLength()).withPosition(1, 0);
+		armStateWidget.addString("Elbow State", () -> Elbow.getInstance().getState().toString()).withPosition(0, 1);
+		armStateWidget.addDouble("Angle", () -> Elbow.getInstance().getAngle()).withPosition(1, 1);
+
+		//arm state
+		armTab.addString("Arm state", () -> "doesn't exist").withPosition(4, 2).withSize(1, 2);
+
+
+		//extender length
+		armTab.addDouble("Extender length", () -> Extender.getInstance().getLength());
+
+		//extender state
+		armTab.addString("Extender state", () -> String.valueOf(Extender.getInstance().getState()));
+
+		//extender ff
+		armTab.addDouble("Extender ff", () -> Extender.getInstance().getFeedForward());
+
+		//moves to length from dashboard input
+		GenericEntry length = armTab.add("Desired length", 0).getEntry();
+		new ExtendToLength(length.getDouble(0)).schedule();
+
+		//extender kp
```

Comment:
I think this code might over-write itself - when you call the I part you erase the p and ...